src/main/java/com/synacy/gradprogram/spock/exercise/OrderingService.java

```diff
@@ -66,8 +68,13 @@ public Order createAnOrder(Cart cart, String recipientName, String recipientAddr
     return order;
   }
 
-  public void cancelOrder(CancelOrderRequest request) {
-    // TODO: Implement me. Cancels PENDING and FOR_DELIVERY orders and create a refund request saving it to the database.
-    //  Else throws an UnableToCancelException
+  public void cancelOrder(CancelOrderRequest request, Order order) throws UnableToCancelException {
+    if (order.getStatus() == OrderStatus.PENDING || order.getStatus() == OrderStatus.FOR_DELIVERY) {
+      order.setStatus(OrderStatus.CANCELLED);
+      orderRepository.saveOrder(order);
+    } else {
+      throw new UnableToCancelException();
+    }
+    refundService.calculateRefund(request, order);
```

Comment:
The `calculateRefund` should only be calculating the refund value. I understand that the `createAndSaveRefundRequest` method in the `RefundService` is private, but what is stopping us from updating it to public if it is needed?